app/actors/sufia/create_with_files_actor.rb

```
@@ -0,0 +1,42 @@
+module Sufia
+  # Creates a work and attaches files to the work
+  class CreateWithFilesActor
+    attr_reader :work_actor, :uploaded_file_ids
+
+    def initialize(work_actor, uploaded_file_ids)
+      @work_actor = work_actor
+      @uploaded_file_ids = uploaded_file_ids
+    end
+
+    delegate :visibility_changed?, to: :work_actor
+
+    def create
+      validate_files && work_actor.create && attach_files
```

Should we be erroring out or at least logging a message if either or both of these `&&`s fail? I'm wondering about silent failures here and in `#update`